### PR TITLE
[Bugfix][Higgs-Audio-V3] Propagate Sampling Parameters to Sampler

### LINKED ISSUE #5387: [Bug]: Higgs Audio v3 (TTS) Sampling Parameters ignored in sampling of audio codes
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Collecting environment information...
uv is set
INFO 07-24 22:51:22 [patch.py:252] NVFP4 W4A4 weight_scale NaN-clamp: installed.
==============================
        System Info
==============================
OS                           : Ubuntu 22.04.5 LTS (x86_64)
GCC version                  : (Ubuntu 11.4.0-1ubuntu1~22.04.3) 11.4.0
Clang version                : Could not collect
CMake version                : version 3.22.1
Libc version                 : glibc-2.35

==============================
       PyTorch Info
==============================
PyTorch version              : 2.11.0+cu129
Is debug build               : False
CUDA used to build PyTorch   : 12.9
ROCM used to build PyTorch   : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.13 (main, Jul 18 2026, 17:02:19) [Clang 22.1.3 ] (64-bit runtime)
Python platform              : Linux-5.15.0-177-generic-x86_64-with-glibc2.35

==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : 12.6.85
CUDA_MODULE_LOADING set to   : 
GPU models and configuration : GPU 0: NVIDIA A100-SXM4-80GB
Nvidia driver version        : 560.35.05
cuDNN version                : Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.7.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.7.1
HIP runtime version          : N/A
MIOpen runtime version       : N/A
Is XNNPACK available         : True

==============================
          CPU Info
==============================
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           43 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  128
On-line CPU(s) list:                     0-127
Vendor ID:                               AuthenticAMD
Model name:                              AMD EPYC 7713 64-Core Processor
CPU family:                              25
Model:                                   1
Thread(s) per core:                      1
Core(s) per socket:                      64
Socket(s):                               2
Stepping:                                1
Frequency boost:                         enabled
CPU max MHz:                             3720.7029
CPU min MHz:                             1500.0000
BogoMIPS:                                3992.40
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca sme sev sev_es ibpb_exit_to_user
Virtualization:                          AMD-V
L1d cache:                               4 MiB (128 instances)
L1i cache:                               4 MiB (128 instances)
L2 cache:                                64 MiB (128 instances)
L3 cache:                                512 MiB (16 instances)
NUMA node(s):                            2
NUMA node0 CPU(s):                       0-63
NUMA node1 CPU(s):                       64-127
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Mitigation; safe RET
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP disabled; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Vulnerable: Clear CPU buffers attempted, no microcode
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

==============================
Versions of relevant libraries
==============================
[pip3] flashinfer-python==0.6.12
[pip3] numpy==2.3.5
[pip3] nvidia-cublas-cu12==12.9.1.4
[pip3] nvidia-cuda-cccl-cu12==12.9.27
[pip3] nvidia-cuda-crt==13.3.73
[pip3] nvidia-cuda-cupti-cu12==12.9.79
[pip3] nvidia-cuda-nvcc==13.2.86
[pip3] nvidia-cuda-nvcc-cu12==12.9.86
[pip3] nvidia-cuda-nvrtc-cu12==12.9.86
[pip3] nvidia-cuda-runtime==13.3.29
[pip3] nvidia-cuda-runtime-cu12==12.9.79
[pip3] nvidia-cuda-tileiras==13.2.86
[pip3] nvidia-cudnn-cu12==9.17.1.4
[pip3] nvidia-cudnn-frontend==1.26.0
[pip3] nvidia-cufft-cu12==11.4.1.4
[pip3] nvidia-cufile-cu12==1.14.1.1
[pip3] nvidia-curand-cu12==10.3.10.19
[pip3] nvidia-cusolver-cu12==11.7.5.82
[pip3] nvidia-cusparse-cu12==12.5.10.65
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-cutlass-dsl==4.5.2
[pip3] nvidia-cutlass-dsl-libs-base==4.5.2
[pip3] nvidia-ml-py==13.610.43
[pip3] nvidia-nccl-cu12==2.28.9
[pip3] nvidia-nvjitlink==13.3.33
[pip3] nvidia-nvjitlink-cu12==12.9.86
[pip3] nvidia-nvshmem-cu12==3.4.5
[pip3] nvidia-nvtx-cu12==12.9.79
[pip3] nvidia-nvvm==13.2.86
[pip3] onnxruntime==1.27.0
[pip3] pyzmq==27.1.0
[pip3] tokenspeed-triton==3.8.10.post20260721
[pip3] torch==2.11.0+cu129
[pip3] torch-c-dlpack-ext==0.1.5
[pip3] torch-einops-utils==0.1.8
[pip3] torchaudio==2.11.0+cu129
[pip3] torchsde==0.2.6
[pip3] torchvision==0.26.0+cu129
[pip3] transformers==5.14.1
[pip3] triton==3.6.0
[pip3] x-transformers==2.24.0
[conda] Could not collect

==============================
         vLLM Info
==============================
ROCM Version                 : Could not collect
vLLM Version                 : 0.24.0
vLLM-Omni Version            : 0.24.0
vLLM Build Flags:
  CUDA Archs: Not Set; ROCm: Disabled
GPU Topology:
        GPU0    NIC0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NODE    65-72   1               N/A
NIC0    NODE     X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0

==============================
     Environment Variables
==============================
CUDA_CACHE_PATH=/home/$USER/.cache/nv
VLLM_ENABLE_V1_MULTIPROCESSING=0
CUDA_VISIBLE_DEVICES=0
VLLM_BATCH_INVARIANT=1
CUBLAS_WORKSPACE_CONFIG=:4096:8
LD_LIBRARY_PATH=/home/$USER/py312/lib/python3.12/site-packages/cv2/../../lib64:/usr/local/cuda-12.6/lib64
OMP_NUM_THREADS=8
CUDA_HOME=/usr/local/cuda-12.6
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_$USER

```

</details>

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
Name: vllm
Version: 0.24.0+cu129

Installed from vllm precomiled wheel directly downloaded from github releases
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
Name: vllm-omni
Version: 0.24.0

```
</details>

### 🐛 Describe the bug

Higgs Audio v3 (TTS) does not take into account sampling parameters passed to the request, in my case by Offline Inference with Voice Cloning, but probably in all cases, for sampling of audio codes. This, in my case, results in VERY similar result no matter which sampling params I pass.

After some research into the model implementation I assume this to be because in [_sample_audio_codes](https://github.com/vllm-project/vllm-omni/blob/d6240742f365cd813991b540b8ea797347aac72e/vllm_omni/model_executor/models/higgs_audio_v3/higgs_audio_v3_talker.py#L1340) it uses hardcoded sampling values that ignore all request set parameters entirely, using hardcoded top_k = 50 and top_p = 0.95, while ignoring temperature entirely (eq. to temperature = 1.0).

Specifically this is in an environment with various reproducibility envs enables and set, which is proably why I noticed that major differences in sampling parameters barely (as in basically not at all) change my eval results.

It's kind of hard to prove this with a short reproducible example but the code reference above should be the best "proof" that it is in fact using hard coded sampling params instead of ones passed.

Specifically I am overwriting the sampling parameters like this with a SamplingParams object for stage 0 (talker)  and passing it to the omni's generate function like this `omni.generate(prompts, sampling_params_list=self._sampling)` but I am confident this isn't the error source.

```python
gen_config = {"temperature": ..., "top_p": ..., "top_k": ...}

stage0 = { # from higgs yaml defaults
            "temperature": 1.0,
            "top_p": 0.95,
            "top_k": 50,
            "max_tokens": 2048,
            "seed": seed,
            "repetition_penalty": 1.0,
            "detokenize": False
        }
        stage0.update(gen_config)

        sampling = [
            SamplingParams(**stage0),
            SamplingParams( # From higgs yaml
                temperature=0.0,
                top_p=1.0,
                top_k=-1,
                max_tokens=65536,
                seed=seed,
                repetition_penalty=1.0,
                detokenize=False
            ),
        ]
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by sphinxkkkbc**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose
Fix #5387

Enable per-request sampling parameters for Higgs Audio v3 speech generation.

- Promotes `temperature`, `top_p`, and `top_k` from speech request `extra_params` into the stage-0 `SamplingParams`.
- Allows vLLM to propagate these values through `InputBatch` into `SamplingMetadata`.
- Passes `SamplingMetadata` to the Higgs Audio v3 codebook sampler.
- Expands request-level sampling parameters from `[batch]` to `[batch * num_vq]`, so every VQ logits row uses the parameters of its owning request.

## Test Plan
- Validate speech request sampling parameters are propagated through stage-0 `SamplingParams` and `SamplingMetadata` to the model-owned `sample()` method.
- Validate Higgs Audio v3 correctly applies per-request `temperature`, `top_p`, and `top_k` across batched requests and their VQ logits, top_k=-1 and invalid_temperature test
```
pytest -v tests/unit/higgs_audio_v3/test_higgs_audio_v3.py
pytest -v tests/worker/test_gpu_ar_model_runner.py 
```
**vLLM Version:**
0.25.0
**vLLM-Omni Commit:**
b57b76a8c1260ca72ce637301f82cd266cbf4090
## Test Result
- test_gpu_ar_model_runner: 21 passed
- test_higgs_audio_v3: 77 passed

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/model_executor/models/higgs_audio_v3/higgs_audio_v3_talker.py [hunk 1]
```diff
@@ -1337,23 +1343,41 @@ def _update_delay_state_batched(
...
-            sorted_logits, sorted_idx = x.sort(dim=-1, descending=True)
-            cumprobs = sorted_logits.softmax(dim=-1).cumsum(dim=-1)
-            sorted_mask = cumprobs > top_p
-            sorted_mask[..., 1:] = sorted_mask[..., :-1].clone()
-            sorted_mask[..., 0] = False
-            mask = torch.zeros_like(x, dtype=torch.bool)
-            mask.scatter_(-1, sorted_idx, sorted_mask)
-            x = x.masked_fill(mask, float("-inf"))
+        temperatures, top_ks, top_ps = self._expand_audio_sampling_params(sampling_metadata, x, num_codebooks)
+        top_ks = top_ks.clamp(min=1, max=x.shape[-1])
```
Comment (Gaohan123) [inline vllm_omni/model_executor/models/higgs_audio_v3/higgs_audio_v3_talker.py]: `top_k=-1` is commonly used by vLLM callers to mean “disable top-k filtering,” but `clamp(min=1, ...)` converts it to `1`. That makes audio-code sampling greedy for those requests, rather than leaving the distribution unrestricted.

Could we preserve the no-top-k sentinel before calling `apply_top_k_top_p`—for example, map `top_k <= 0` to `x.shape[-1]` (or the helper’s documented disable value)? Please also add a regression test covering `top_k=-1`, ensuring more than one candidate remains sampleable.

Comment (sphinxkkkbc) [reply vllm_omni/model_executor/models/higgs_audio_v3/higgs_audio_v3_talker.py]: fixed, test added and descriptions updated

**Review by Gaohan123 — APPROVED** [review]
LGTM. Thanks
